### PR TITLE
VFB-126: Replace auth component

### PR DESCRIPTION
**Pull request by rickaw**
## What's changed
Replace the library auth component with our own custom-made component.

#### Sign in
image

#### Password reset
image

#### Sign in
https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/assets/114006382/951e391f-bd8e-461a-a21f-9e195689011b

#### Password reset
https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/assets/114006382/bb86e252-49ab-4010-a2e5-d7828f2c39a0

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA 
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
  - Will also be automatically checked on pull request
- [x] Make sure you've tested via `npm run test`
  - Will also be automatically checked on pull request

### CODE REVIEW

File: src/components/AuthPanel.tsx
Comment (rickaw) [inline src/components/AuthPanel.tsx:1]: This file was modified from LoginPanel.tsx, replacing the library component with our own component.

File: src/components/AuthPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        color: ${(props) => props.theme.primary.foreground[2]};
+    }
+`;
+
+interface AuthPanelProps {
+    title: string;
+    emailField: AuthTextField | null;
+    passwordField: AuthTextField | null;
+    submitText: string;
+    onSubmit: (() => void) | (() => Promise<void>);
```
Comment (rickaw) [inline src/components/AuthPanel.tsx:53]: no need for promise

File: src/components/AuthPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                        id="email"
+                        label="Email address"
+                        variant="outlined"
+                        value={emailField.text}
+                        onChange={(event) => emailField.setText(event.target.value)}
+                    />
+                )}
+
+                {passwordField && (
+                    <TextField
```
Comment (rickaw) [inline src/components/AuthPanel.tsx:97]: hide the password text

File: src/components/AuthPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    errorMessage,
+    successMessage,
+}) => {
+    const theme = useTheme();
+
+    return (
+        <MiddleDiv elevation={5} id="login-panel">
+            <Title>{title}</Title>
+
+            <div style={{ display: "flex", flexDirection: "column", gap: "10px" }}>
```
Comment (rickaw) [inline src/components/AuthPanel.tsx:85]: remove inline style

File: src/components/AuthPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                            href={authLink.href}
+                            style={{
+                                color: theme.main.lighterForeground[0],
+                                fontSize: "13px",
+                                textAlign: "center",
+                            }}
+                        >
+                            {authLink.label}
+                        </Link>
+                    ))}
```
Comment (rickaw) [inline src/components/AuthPanel.tsx:123]: } next line

File: src/app/auth/callback/route.ts
Comment (rickaw) [inline src/app/auth/callback/route.ts:1]: This route handler is now unused.

**Review by stonelink — APPROVED** [review]
Thank you for doing this. Sorry it's taken me a couple of days to get to it.

Is it necessary for the login panel to become so much narrower? It doesn't look like it would wide enough to show enough of any reasonable email address for the user to have that sanity check of 'did I type it right?'

**Comment by rickaw** [comment]
> n't look like it would wide enough to show enough of any reasonable email address for the user to have that sanity check of 'did

really good point - I've changed it now so that the auth panel can grow bigger a little larger

**Pull request merged**